pkg/operator/staticpod/controller/node/node_controller_test.go

```diff
@@ -48,122 +51,403 @@ func addNodeReadyCondition(node *corev1.Node, status corev1.ConditionStatus) *co
 
-				con := conditions[0]
-				if err := validateCommonNodeControllerDegradedCondtion(con); err != nil {
-					return err
-				}
-				if con.Status != operatorv1.ConditionTrue {
-					return fmt.Errorf("incorrect condition.status, expected %v, got %v", operatorv1.ConditionTrue, con.Status)
-				}
-				expectedMsg := `The master nodes not ready: node "test-node-1" not ready since 2018-01-12 22:51:48.324359102 +0000 UTC because TestReason (test message), node "test-node-3" not ready since 2018-01-12 22:51:48.324359102 +0000 UTC because TestReason (test message)`
-				if con.Message != expectedMsg {
-					return fmt.Errorf("incorrect condition.message, expected %s, got %s", expectedMsg, con.Message)
-				}
-				return nil
+				return validateNodeControllerDegradedCondition(conditions, expectedCondition)
 			},
 		},
 
-		// scenario 4: Ready condition not present in status block
 		{
 			name:        "scenario 4: Ready condition not present in status block",
 			masterNodes: []runtime.Object{makeNodeReady(fakeMasterNode("test-node-1")), fakeMasterNode("test-node-2")},
-			evaluateNodeStatus: func(conditions []operatorv1.OperatorCondition) error {
-				if len(conditions) != 1 {
-					return fmt.Errorf("expected exaclty 1 condition, got %d", len(conditions))
-				}
+			verifyNodeStatus: func(conditions []operatorv1.OperatorCondition) error {
+				var expectedCondition operatorv1.OperatorCondition
+				expectedCondition.Type = condition.NodeControllerDegradedConditionType
+				expectedCondition.Reason = "MasterNodesReady"
+				expectedCondition.Status = operatorv1.ConditionTrue
+				expectedCondition.Message = `The master nodes not ready: node "test-node-2" not ready, no Ready condition found in status block`
+
+				return validateNodeControllerDegradedCondition(conditions, expectedCondition)
+			},
+		},
 
-				con := conditions[0]
-				if err := validateCommonNodeControllerDegradedCondtion(con); err != nil {
-					return err
+		{
+			name:        "scenario 5: a JSON patch is created when two master nodes were removed",
+			masterNodes: []runtime.Object{makeNodeReady(fakeMasterNode("test-node-2")), makeNodeReady(fakeMasterNode("test-node-4"))},
+			existingNodeStatuses: []operatorv1.NodeStatus{
+				{
+					NodeName: "test-node-1",
+				},
+				{
+					NodeName: "test-node-2",
+				},
+				{
+					NodeName: "test-node-3",
+				},
+				{
+					NodeName: "test-node-4",
+				},
+			},
+			verifyJSONPatch: func(actualJSONPatch *jsonpatch.PatchSet) error {
+				expectedJSONPatch := jsonpatch.New().
+					WithRemove("/status/nodeStatuses/0", jsonpatch.NewTestCondition("/status/nodeStatuses/0/nodeName", "test-node-1")).
+					WithRemove("/status/nodeStatuses/1", jsonpatch.NewTestCondition("/status/nodeStatuses/1/nodeName", "test-node-3"))
```

Comment:
hm, I was confused at first, but this is `1` because the previous `WithRemove` reshuffled the indexes